staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -94,12 +95,26 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 		}
 
 		userInfo, _ := request.UserFrom(ctx)
-		staticAdmissionAttributes := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo)
 		var transformers []rest.TransformFunc
-		if mutatingAdmission, ok := admit.(admission.MutationInterface); ok && mutatingAdmission.Handles(admission.Update) {
+		if mutatingAdmission, ok := admit.(admission.MutationInterface); ok {
 			transformers = append(transformers, func(ctx context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
-				return newObj, mutatingAdmission.Admit(admission.NewAttributesRecord(newObj, oldObj, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
+				if oldObj == nil {
+					return nil, fmt.Errorf("unexpected nil object in mutating admission check of UPDATE request")
+				}
+				accessor, err := meta.Accessor(oldObj)
+				if err != nil {
+					return nil, err
+				}
+				if len(accessor.GetResourceVersion()) == 0 {
```

Comment:
like @jennybuckley mentioned, can we change this check to UID (like patch does)?

https://github.com/kubernetes/kubernetes/blob/d5803e596fc8aba17aa8c74a96aff9c73bb0f1da/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go#L319-L323

Is it guaranteed that the resourceVersion is always 0 (I think it should be, but wondering if we can enforce it).

Reply:
@nikhita Nice catch. Referencing `hasUID` would be more appropriate.